tools/triage/add_triage_signature.py

```diff
@@ -2686,6 +2686,46 @@ def _process_ticket(self, url, issue_key):
             self._process_ticket_helper(url, old_logs_path)
 
 
+class ConfiguringHostAndMachineConfigServerError(ErrorSignature):
+    """
+    Looks for a host that timed out in "Configuring" in addition to errors in the MCS logs
+    """
+
+    mcs_error = re.compile(r"couldn't get config for req")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. machine-config-server failed to serve ignition",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        md = get_metadata_json(url)
+
+        cluster = md["cluster"]
+
+        hosts_in_configuring = False
+        for host in cluster["hosts"]:
+            stage = host["progress"]["current_stage"]
+            logger.debug("found host with current stage {}".format(stage))
+            hosts_in_configuring = hosts_in_configuring or (stage == "Configuring")
+
+        if not hosts_in_configuring:
+            logger.debug("found no hosts in configuring")
+            return
+
+        try:
+            mcs_logs = get_triage_logs_tar(triage_url=url, cluster_id=md["cluster"]["id"]).get(
+                "controller_logs.tar.gz/must-gather.tar.gz/must-gather.local.*/*/namespaces/openshift-machine-config-operator/pods/machine-config-server-*/machine-config-server/machine-config-server/logs/current.log"
```

Comment:
Given https://github.com/omertuc/nestedarchive/issues/2 I think I need to change this.

IIUC this will only give me the output of the first matching file, right?
We probably need to iterate through each replica of the mcs and search the logs for each of them.

Reply:
Right. If you give it the path of a directory you get a temp dir mirroring the dir in the archive